File: PE/437.cpp
#include <bits/stdc++.h>
using namespace std;
int v[100000020];
long long ans = 5;
int cnt = 1;

bool judge(int p) {
	return true;
}
int main() {
	int n = 100;
	for (int i = 2; i <= n; i++) {
		if (v[i] > 0) {
			continue;
		}
		if (i % 10 == 1 || i % 10 == 9) {
			if (judge(i)) {
				ans += i;
				cnt++;
				printf("%d\n", i);
			}
		}
		for (int j = i; j <= n; j += i) {
			v[j] = i;
		}
	}
	cout << cnt << endl;
	cout << ans << endl;
	return 0;
}

File: PE/437.py
from sage.all import *
def R(p):
	A = Integers(p)
	d = A(5)
	if d.is_square():
		x1 = (1 + d.sqrt()) / 2
		x2 = (1 - d.sqrt()) / 2
		if x1.multiplicative_order() == p - 1 or x2.multiplicative_order() == p - 1:
			return True
		else:
			return False
	return False

ans = []
cnt = 0
for i in primes(3, 10 ** 8):
	if R(i):
		ans.append(i)
	cnt += 1
	if cnt % 10000 == 0:
		print cnt, i
print sum(ans), len(ans)